src/huggingface_hub/hf_api.py

```diff
@@ -738,9 +747,11 @@ def _unpack_model_filter(self, model_filter: ModelFilter):
         query_dict["filter"] = tuple(filter_tuple)
         return query_dict
 
+    @_deprecate_positional_args
     def list_datasets(
         self,
         filter: Union[DatasetFilter, str, Iterable[str], None] = None,
+        *,
```

Comment:
I personally wouldn't find it weird to have the `*` be placed right after `self` and right before `filter`. I think `filter` holds as much weight as `author` or `search`, for example.